ngraph/core/include/openvino/core/preprocess/input_tensor_info.hpp

```diff
@@ -117,6 +129,28 @@ class OPENVINO_API InputTensorInfo final {
     ///
     /// \return Rvalue reference to 'this' to allow chaining with other calls in a builder-like manner.
     InputTensorInfo&& set_spatial_static_shape(size_t height, size_t width) &&;
+
+    /// \brief Manually set shape for user's input tensor. In many cases tensor's shape is calculated based on layout,
+    /// color format, spatial dimensions, however user has possibility to manually set of input shape.
+    ///
+    /// This version allows chaining for Lvalue objects
+    ///
+    /// \param shape Array of dimensions representing input shape, e.g. {-1, 3, 224, 224}. Value 0 or -1 indicates
+    /// 'dynamic' dimension
+    ///
+    /// \return Reference to 'this' to allow chaining with other calls in a builder-like manner
+    InputTensorInfo& set_data_shape(const std::vector<int64_t>& shape) &;
+
+    /// \brief Manually set shape for user's input tensor. In many cases tensor's shape is calculated based on layout,
+    /// color format, spatial dimensions, however user has possibility to manually set of input shape.
+    ///
+    /// This version allows chaining for Rvalue objects
+    ///
+    /// \param shape Array of dimensions representing input shape, e.g. {-1, 3, 224, 224}. Value 0 or -1 indicates
+    /// 'dynamic' dimension
+    ///
+    /// \return Rvalue reference to 'this' to allow chaining with other calls in a builder-like manner
+    InputTensorInfo&& set_data_shape(const std::vector<int64_t>& shape) &&;
```

Comment:
it may confuse `NHWC` with `[1 3 224 224]` because we have 1 more level of freedom

Reply:
we can possibly have one method like `InputTensorInfo().set_layout("NCHW", {1, 3, 224, 224}) if it looks more convenient

Reply:
Removed `set_data_shape`